### ISSUE TITLE
NullPointerException happened when use checker/total-queue

### ISSUE THREAD

**Issue by RongtongJin** [jepsen-io/jepsen#414]
Hello, I'm testing the message queue with jepsen .
I used checker/total-queue to check for consistency, but there was a NullPointerException.
This is the exception stack.
```
INFO [2019-09-13 22:50:16,474] jepsen test runner - jepsen.core Analyzing...
INFO [2019-09-13 22:50:16,487] jepsen test runner - jepsen.core Analysis complete
INFO [2019-09-13 22:50:16,499] jepsen results - jepsen.store Wrote /root/jrt/rocketmq-jepsen-test/store/rocketmq-jepsen-test/20190913T224810.000+0800/results.edn
INFO [2019-09-13 22:50:16,500] jepsen node 172.16.2.123 - rocketmq-jepsen-test.core Stop rocketmq broker 172.16.2.123
INFO [2019-09-13 22:50:16,500] jepsen node 172.16.2.122 - rocketmq-jepsen-test.core Stop rocketmq broker 172.16.2.122
INFO [2019-09-13 22:50:16,500] jepsen node 172.16.2.121 - rocketmq-jepsen-test.core Stop rocketmq broker 172.16.2.121
INFO [2019-09-13 22:50:16,500] jepsen node 172.16.2.127 - rocketmq-jepsen-test.core Stop rocketmq broker 172.16.2.127
INFO [2019-09-13 22:50:16,500] jepsen node 172.16.2.124 - rocketmq-jepsen-test.core Stop rocketmq broker 172.16.2.124
INFO [2019-09-13 22:50:16,638] NettyClientSelector_1 - RocketmqRemoting closeChannel: close the connection to remote address[172.16.2.127:30911] result: true
ERROR [2019-09-13 22:50:36,730] main - jepsen.cli Oh jeez, I'm sorry, Jepsen broke. Here's why:
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[na:1.8.0_211]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[na:1.8.0_211]
	at clojure.core$deref_future.invokeStatic(core.clj:2300) ~[clojure-1.10.0.jar:na]
	at clojure.core$future_call$reify__8439.deref(core.clj:6974) ~[clojure-1.10.0.jar:na]
	at clojure.core$deref.invokeStatic(core.clj:2320) ~[clojure-1.10.0.jar:na]
	at clojure.core$deref.invoke(core.clj:2306) ~[clojure-1.10.0.jar:na]
	at clojure.core$map$fn__5851.invoke(core.clj:2753) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RT.seq(RT.java:531) ~[clojure-1.10.0.jar:na]
	at clojure.core$seq__5387.invokeStatic(core.clj:137) ~[clojure-1.10.0.jar:na]
	at clojure.core$dorun.invokeStatic(core.clj:3133) ~[clojure-1.10.0.jar:na]
	at clojure.core$dorun.invoke(core.clj:3133) ~[clojure-1.10.0.jar:na]
	at jepsen.store$save_2_BANG_.invokeStatic(store.clj:384) ~[jepsen-0.1.11.jar:na]
	at jepsen.store$save_2_BANG_.invoke(store.clj:380) ~[jepsen-0.1.11.jar:na]
	at jepsen.core$analyze_BANG_.invokeStatic(core.clj:450) ~[jepsen-0.1.11.jar:na]
	at jepsen.core$analyze_BANG_.invoke(core.clj:434) ~[jepsen-0.1.11.jar:na]
	at jepsen.core$run_BANG_$fn__6191.invoke(core.clj:565) ~[jepsen-0.1.11.jar:na]
	at jepsen.core$run_BANG_.invokeStatic(core.clj:539) ~[jepsen-0.1.11.jar:na]
	at jepsen.core$run_BANG_.invoke(core.clj:467) ~[jepsen-0.1.11.jar:na]
	at jepsen.cli$single_test_cmd$fn__6896.invoke(cli.clj:362) ~[jepsen-0.1.11.jar:na]
	at jepsen.cli$run_BANG_.invokeStatic(cli.clj:299) [jepsen-0.1.11.jar:na]
	at jepsen.cli$run_BANG_.invoke(cli.clj:229) [jepsen-0.1.11.jar:na]
	at rocketmq_jepsen_test.core$_main.invokeStatic(core.clj:180) [na:na]
	at rocketmq_jepsen_test.core$_main.doInvoke(core.clj:176) [na:na]
	at clojure.lang.RestFn.invoke(RestFn.java:551) [clojure-1.10.0.jar:na]
	at clojure.lang.Var.invoke(Var.java:424) [clojure-1.10.0.jar:na]
	at user$eval140.invokeStatic(form-init2217618709812673373.clj:1) [na:na]
	at user$eval140.invoke(form-init2217618709812673373.clj:1) [na:na]
	at clojure.lang.Compiler.eval(Compiler.java:7176) [clojure-1.10.0.jar:na]
	at clojure.lang.Compiler.eval(Compiler.java:7166) [clojure-1.10.0.jar:na]
	at clojure.lang.Compiler.load(Compiler.java:7635) [clojure-1.10.0.jar:na]
	at clojure.lang.Compiler.loadFile(Compiler.java:7573) [clojure-1.10.0.jar:na]
	at clojure.main$load_script.invokeStatic(main.clj:452) [clojure-1.10.0.jar:na]
	at clojure.main$init_opt.invokeStatic(main.clj:454) [clojure-1.10.0.jar:na]
	at clojure.main$init_opt.invoke(main.clj:454) [clojure-1.10.0.jar:na]
	at clojure.main$initialize.invokeStatic(main.clj:485) [clojure-1.10.0.jar:na]
	at clojure.main$null_opt.invokeStatic(main.clj:519) [clojure-1.10.0.jar:na]
	at clojure.main$null_opt.invoke(main.clj:516) [clojure-1.10.0.jar:na]
	at clojure.main$main.invokeStatic(main.clj:598) [clojure-1.10.0.jar:na]
	at clojure.main$main.doInvoke(main.clj:561) [clojure-1.10.0.jar:na]
	at clojure.lang.RestFn.applyTo(RestFn.java:137) [clojure-1.10.0.jar:na]
	at clojure.lang.Var.applyTo(Var.java:705) [clojure-1.10.0.jar:na]
	at clojure.main.main(main.java:37) [clojure-1.10.0.jar:na]
Caused by: java.lang.NullPointerException: null
	at multiset.core.MultiSet.iterator(core.clj:73) ~[knossos-0.3.3.jar:na]
	at clojure.lang.RT.iter(RT.java:569) ~[clojure-1.10.0.jar:na]
	at clojure.core$sequence.invokeStatic(core.clj:2663) ~[clojure-1.10.0.jar:na]
	at clojure.core$sequence.invoke(core.clj:2647) ~[clojure-1.10.0.jar:na]
	at fipp.edn$pretty_coll.invokeStatic(edn.cljc:15) ~[jepsen-0.1.11.jar:na]
	at fipp.edn$pretty_coll.invoke(edn.cljc:8) ~[jepsen-0.1.11.jar:na]
	at fipp.edn.EdnPrinter.visit_set(edn.cljc:65) ~[jepsen-0.1.11.jar:na]
	at fipp.visit$visit_STAR_.invokeStatic(visit.cljc:48) ~[jepsen-0.1.11.jar:na]
	at fipp.visit$visit_STAR_.invoke(visit.cljc:32) ~[jepsen-0.1.11.jar:na]
	at fipp.visit$visit.invokeStatic(visit.cljc:61) ~[jepsen-0.1.11.jar:na]
	at fipp.visit$visit.invoke(visit.cljc:58) ~[jepsen-0.1.11.jar:na]
	at fipp.edn.EdnPrinter$fn__2520.invoke(edn.cljc:62) ~[jepsen-0.1.11.jar:na]
	at fipp.edn$pretty_coll$fn__2499.invoke(edn.cljc:12) ~[jepsen-0.1.11.jar:na]
	at clojure.core$map$fn__5847$fn__5848.invoke(core.clj:2742) ~[clojure-1.10.0.jar:na]
	at clojure.lang.TransformerIterator.step(TransformerIterator.java:79) ~[clojure-1.10.0.jar:na]
	at clojure.lang.TransformerIterator.hasNext(TransformerIterator.java:97) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RT$4.invoke(RT.java:518) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RT.seq(RT.java:531) ~[clojure-1.10.0.jar:na]
	at clojure.core$seq__5387.invokeStatic(core.clj:137) ~[clojure-1.10.0.jar:na]
	at clojure.core$map$fn__5851.invoke(core.clj:2746) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RT.seq(RT.java:531) ~[clojure-1.10.0.jar:na]
	at clojure.core$seq__5387.invokeStatic(core.clj:137) ~[clojure-1.10.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:660) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.invokeStatic(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.doInvoke(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RestFn.invoke(RestFn.java:423) [clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:15) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at clojure.core$map$fn__5851.invoke(core.clj:2753) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RT.seq(RT.java:531) ~[clojure-1.10.0.jar:na]
	at clojure.core$seq__5387.invokeStatic(core.clj:137) ~[clojure-1.10.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:660) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.invokeStatic(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.doInvoke(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RestFn.invoke(RestFn.java:423) [clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:15) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$fn__2459.invokeStatic(engine.cljc:63) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$fn__2459.invoke(engine.cljc:58) ~[jepsen-0.1.11.jar:na]
	at clojure.lang.MultiFn.invoke(MultiFn.java:229) ~[clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:18) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at clojure.core$map$fn__5851.invoke(core.clj:2753) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RT.seq(RT.java:531) ~[clojure-1.10.0.jar:na]
	at clojure.core$seq__5387.invokeStatic(core.clj:137) ~[clojure-1.10.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:660) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.invokeStatic(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.doInvoke(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RestFn.invoke(RestFn.java:423) [clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:15) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$fn__2444.invokeStatic(engine.cljc:48) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$fn__2444.invoke(engine.cljc:47) ~[jepsen-0.1.11.jar:na]
	at clojure.lang.MultiFn.invoke(MultiFn.java:229) ~[clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:18) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at clojure.core$map$fn__5851.invoke(core.clj:2753) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RT.seq(RT.java:531) ~[clojure-1.10.0.jar:na]
	at clojure.core$seq__5387.invokeStatic(core.clj:137) ~[clojure-1.10.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:660) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.invokeStatic(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.doInvoke(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RestFn.invoke(RestFn.java:423) [clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:15) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$fn__2428.invokeStatic(engine.cljc:35) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$fn__2428.invoke(engine.cljc:34) ~[jepsen-0.1.11.jar:na]
	at clojure.lang.MultiFn.invoke(MultiFn.java:229) ~[clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:18) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at clojure.core$map$fn__5851.invoke(core.clj:2753) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RT.seq(RT.java:531) ~[clojure-1.10.0.jar:na]
	at clojure.core$seq__5387.invokeStatic(core.clj:137) ~[clojure-1.10.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:660) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.invokeStatic(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.doInvoke(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RestFn.invoke(RestFn.java:423) [clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:15) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at clojure.core$map$fn__5851.invoke(core.clj:2753) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RT.seq(RT.java:531) ~[clojure-1.10.0.jar:na]
	at clojure.core$seq__5387.invokeStatic(core.clj:137) ~[clojure-1.10.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:660) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.invokeStatic(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.doInvoke(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RestFn.invoke(RestFn.java:423) [clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:15) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$fn__2459.invokeStatic(engine.cljc:63) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$fn__2459.invoke(engine.cljc:58) ~[jepsen-0.1.11.jar:na]
	at clojure.lang.MultiFn.invoke(MultiFn.java:229) ~[clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:18) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at clojure.core$map$fn__5851.invoke(core.clj:2753) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.sval(LazySeq.java:42) ~[clojure-1.10.0.jar:na]
	at clojure.lang.LazySeq.seq(LazySeq.java:51) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RT.seq(RT.java:531) ~[clojure-1.10.0.jar:na]
	at clojure.core$seq__5387.invokeStatic(core.clj:137) ~[clojure-1.10.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:660) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.invokeStatic(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.core$mapcat.doInvoke(core.clj:2783) ~[clojure-1.10.0.jar:na]
	at clojure.lang.RestFn.invoke(RestFn.java:423) [clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:15) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$fn__2444.invokeStatic(engine.cljc:48) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$fn__2444.invoke(engine.cljc:47) ~[jepsen-0.1.11.jar:na]
	at clojure.lang.MultiFn.invoke(MultiFn.java:229) ~[clojure-1.10.0.jar:na]
	at fipp.engine$serialize.invokeStatic(engine.cljc:18) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$serialize.invoke(engine.cljc:12) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$pprint_document.invokeStatic(engine.cljc:236) ~[jepsen-0.1.11.jar:na]
	at fipp.engine$pprint_document.invoke(engine.cljc:234) ~[jepsen-0.1.11.jar:na]
	at fipp.edn$pprint.invokeStatic(edn.cljc:100) ~[jepsen-0.1.11.jar:na]
	at fipp.edn$pprint.invoke(edn.cljc:91) ~[jepsen-0.1.11.jar:na]
	at fipp.edn$pprint.invokeStatic(edn.cljc:92) ~[jepsen-0.1.11.jar:na]
	at fipp.edn$pprint.invoke(edn.cljc:91) ~[jepsen-0.1.11.jar:na]
	at jepsen.store$write_results_BANG_.invokeStatic(store.clj:344) ~[jepsen-0.1.11.jar:na]
	at jepsen.store$write_results_BANG_.invoke(store.clj:340) ~[jepsen-0.1.11.jar:na]
	at jepsen.store$save_2_BANG_$fn__4494.invoke(store.clj:384) ~[jepsen-0.1.11.jar:na]
	at clojure.core$binding_conveyor_fn$fn__5739.invoke(core.clj:2030) ~[clojure-1.10.0.jar:na]
	at clojure.lang.AFn.call(AFn.java:18) ~[clojure-1.10.0.jar:na]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_211]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_211]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_211]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_211]

```
Can someone help me?

**Comment by aphyr** [comment]
IIRC this was a bug in the Multiset library, which was fixed by upgrading that dependency in Jepsen a few versions ago. What version of Jepsen are you running?

**Comment by RongtongJin** [comment]
@aphyr Thanks for your reply, I tried jepsen 0.1.15 (the latest realse), but it not worked.

**Comment by aphyr** [comment]
OK! Perhaps there's another issue in multiset.core.MultiSet.iterator then--I'd suggest starting your search there. :)

**Comment by aphyr** [comment]
(here's the previous issue, if it's helpful: https://github.com/jepsen-io/jepsen/issues/314)

**Comment by RongtongJin** [comment]
@aphyr Thanks, it‘s helpful. This bug was already fixed in [knossos pr #30](https://github.com/jepsen-io/knossos/pull/30). I will close this issue.

**Comment by aphyr** [comment]
I was waiting for more changes to accrue on knossos before cutting a release, but yeah, if it's shadowing the version in Jepsen we need to get that done! I've cut Knossos 0.3.5 and released that; you should be able to add that dependency explicitly to your project.clj (and exclude it from jepsen) to fix this problem.
